src/main/java/org/opensearch/security/configuration/SecurityIndexSearcherWrapper.java

```diff
@@ -168,4 +177,13 @@ protected final boolean isPermittedOnIndex() {
         }
         return false;
     }
+
+    private boolean isPermittedOnSystemIndex() {
+        final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+        final TransportAddress caller = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
+        final Set<String> mappedRoles = evaluator.mapRoles(user, caller);
+        final SecurityRoles securityRoles = evaluator.getSecurityRoles(mappedRoles);
+
+        return securityRoles.isPermittedOnSystemIndex(index.getName());
```

Comment:
is there a way we can leverage `hasExplicitIndexPermission` here?

Reply:
I want it. But I cannot find a way to get the parameters `Resolved, IndexNameExpressionResolver` for `hasExplicitIndexPermission`. Could I get `Resolved, IndexNameExpressionResolver` at that location?

Reply:
It might be tricky to get it there as there are no direct references that you can use.